### PR TITLE
Set active scalars in data sources

### PR DESCRIPTION
**Pull request by cquammen**
This branch adds initial support for selecting the "active" scalars in a data set. The active scalars are displayed in the histogram, used for coloring in different modules, and modified by transforms.

To select the active array, click on the data set. The list of arrays will appear as in the figure below:

image

Click on the row in the table of arrays to set that array as active. In this example, the array 'coordsX' is active.

We do not need to keep the selection behavior in the UI - opinions on ways to set the active array are welcome (combo box with array names, for instance?).

### CODE REVIEW

File: tomviz/DataSource.cxx [hunk 1]
```diff
@@ -497,6 +497,40 @@ void DataSource::setSpacing(const double spacing[3])
   emit dataPropertiesChanged();
 }
 
+void DataSource::setActiveScalars(const char* arrayName)
```
Comment (cryos) [inline tomviz/DataSource.cxx]: I feel like the API would fit better if it used QString, and then created the const char* for the VTK function.

File: tomviz/DataSource.cxx [hunk 2]
```diff
@@ -497,6 +497,40 @@ void DataSource::setSpacing(const double spacing[3])
...
+    }
+  }
+
+  dataModified();
+
+  emit activeScalarsChanged();
+  emit dataPropertiesChanged();
+}
+
+const char* DataSource::activeScalars() const
```
Comment (cryos) [inline tomviz/DataSource.cxx]: Same here on the return type

File: tomviz/DataSource.cxx (hunk 1 above)
Comment (cquammen) [reply tomviz/DataSource.cxx]: Agreed. Changed in latest revision.

**Comment by cquammen** [comment]
Here's a test data file: https://data.kitware.com/#item/5a57b72d8d777f5e872f8380

**Pull request merged**